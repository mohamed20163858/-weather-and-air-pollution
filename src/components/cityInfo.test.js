import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import CityInfo from './cityInfo';

function TestCityInfo() {
  return (
    <Provider store={store}>
      <Router>
        <CityInfo country="Egypt" city="Cairo" />
      </Router>
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestCityInfo />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('test the existence of country field', () => {
  // ARRANGE
  render(<TestCityInfo />);

  // ACT
  // Arrange
  const result = screen.getByText('Country: Egypt');
  // ASSERT
  expect(result).toBeInTheDocument();
});
