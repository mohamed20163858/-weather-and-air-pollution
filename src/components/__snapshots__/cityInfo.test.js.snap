// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="city-positions-details"
  >
    <p>
      Country: Egypt
    </p>
    <p>
      City: Cairo
    </p>
    <p>
      Latitude: 0
      <sup>
        o
      </sup>
    </p>
    <p>
      Longitude: 0
      <sup>
        o
      </sup>
    </p>
  </div>,
  <div
    className="weather-analytics"
  >
    <h2>
      Current Weather Analytics
    </h2>
    <p>
      Weather condition: undefined
    </p>
    <p>
      Humidity: undefined %
    </p>
    <p>
      Temperature: undefined
      <sup>
        o
      </sup>
      C
    </p>
    <p>
      Feels like: undefined
      <sup>
        o
      </sup>
      C
    </p>
    <p>
      Cloudness: undefined %
    </p>
    <p>
      Wind speed: undefined m/s
    </p>
    <p>
      Wind degree: undefined
      <sup>
        o
      </sup>
    </p>
    <p>
      Wind gust speed: undefined m/s
    </p>
  </div>,
  <div
    className="air-pollution-analytics"
  >
    <h2>
      Current Air Pollution Analytics
    </h2>
    <p>
      Pollution level: undefined
    </p>
    <p>
      Pollution index: undefined
    </p>
    <h3>
      Concentartions
    </h3>
    <p>
      Carbon monoxide (CO): undefined ug/m3
    </p>
    <p>
      Nitrogen monoxide (NO): undefined ug/m3
    </p>
    <p>
      Nitrogen dioxide (NO2): undefined ug/m3
    </p>
    <p>
      Ozone (O3): undefined ug/m3
    </p>
    <p>
      Sulphur dioxide (SO2): undefined ug/m3
    </p>
    <p>
      Ammonia (NH3): undefined ug/m3
    </p>
    <p>
      Airbrone particles (PM2.5): undefined ug/m3
    </p>
    <p>
      Airbrone particles (PM10): undefined ug/m3
    </p>
  </div>,
]
`;
